pkg/filebacked/file.go

```diff
@@ -41,15 +41,19 @@ type Writer struct {
 	path string
 	// File.
 	file *os.File
-	// Number of objects written.
-	length uint64
+	// Direct access index.
+	index []int64
+	// Dirty (needs flush).
+	dirty bool
```

Comment:
Add index for direct access support.
Add `dirty` to track when the file needs to be flushed.